src/query/plan/rule_based_planner.cpp

```diff
@@ -235,22 +235,11 @@ class ReturnBodyContext : public HierarchicalTreeVisitor {
     return true;
   }
 
-  bool PreVisit(Exists &exists) override {
-    in_exists_ = true;
-    return true;
-  }
-
-  bool PostVisit(Exists &exists) override {
-    in_exists_ = false;
-    return true;
+  bool PreVisit(Exists & /*exists*/) override {
+    throw SemanticException("Exists functionality works only with matching clauses!");
```

Comment:
I see that you introduced error if you try to use exists with WITH clause. I think this shouldn't break on rule based planner but on symbol generator. Is there reason why not throw error earlier. Theoretically, this is "wrong" as you are still letting use unbounded symbols in WITH clause

Reply:
There is part of the code in VisitReturnBody which should trigger deleting variables from scope which are old:

It is probably somewhere here, I tried to research it today:
```
if ((!where && body.order_by.empty()) || scope.has_aggregation) {
    // WHERE and ORDER BY need to see both the old and new symbols, unless we
    // have an aggregation. Therefore, we can clear the symbols immediately if
    // there is neither ORDER BY nor WHERE, or we have an aggregation.
    scope.symbols.clear();
    removed_old_names = true;
  }
```

I think problem happens somewhere here

Reply:
I think we are inconsistent with throwing unbounded variable errors. For example, this is allowed:
```
MATCH (n), (m) 
WITH n 
WHERE id(m)>0 
RETURN n;
```

I think this one should be to:
```
MATCH (n),(m) 
WITH n 
WHERE exists((n)-[]->(m)) 
RETURN n;
```

Reply:
I think this is resolved as we won't fix this.